src/Avalonia.Controls/Documents/InlineUIContainer.cs

```diff
@@ -18,6 +18,8 @@ public class InlineUIContainer : Inline
         public static readonly StyledProperty<Control> ChildProperty =
             AvaloniaProperty.Register<InlineUIContainer, Control>(nameof(Child));
 
+        private double _measuredWidth = double.NaN;
```

Comment:
If the child changes, this should be reset

Reply:
wouldn't changing the child invalidate its measure?